tests/test_resource.py

```diff
@@ -233,28 +161,90 @@ def test_method(self):
             "request_id": ANY,
         }
 
-    def test__get_name__uses_custom_name_attribute(self):
+    def test_get_name_from__name_parameter(self) -> None:
         class XResource(Resource):
             _name = "test"
 
         assert XResource.get_name() == "test"
 
-    def test__get_name__uses_class_name_attribute_when_custom_name_is_not_defined(self):
+    def test_get_name_from__name__parameter(self) -> None:
         class XResource(Resource):
             pass
 
         assert XResource.get_name() == "xresource"
 
+    def test_get_all_possible_authz(self, sample_event) -> None:
+        azc = AuthzCollector()
+        azc.clean()
+
+        class XResource(Resource):
+            _name = "test_auth_collector"
+
+        # import pdb; pdb.set_trace()
+        res = XResource(sample_event)
+        assert res.get_authz_data() == {
+            "test_auth_collector": {
+                "possible_permissions": {},
+                "guest_permissions": {},
+            }
+        }
+
+
+class TestResourceAuthentication:
+    @pytest.fixture(autouse=True)
+    def setup_class(self, sample_event) -> None:
+        class XResource(Resource):
+            @add_route("/")
+            def test_method(self) -> None:
+                return Response("x")
+
+        self.res = XResource  # pylint: disable=attribute-defined-outside-init
```

Comment:
It would be good to name it `self.resource_cls` or something similar as you assigned the whole class to this attribute 😉